src/systemtest/resources/systemtest.sh

```diff
@@ -0,0 +1,165 @@
+#!/bin/bash
+
+# This script should be executed from the root of the repository.
+# Run: src/systemtest/resources/systemtest.sh [COMMAND]
+
+STATUS=0
+
+export CURRENT_DIR=$(pwd)
+echo "Executing script: ${0} from directory: " ${CURRENT_DIR}
+
+function do_build {
+    echo "Building Eiffel Intelligence front-end war file"
+    # Build war file from latest code changes in EI front-end
+    mvn clean -q
+    mvn package -DskipTests=true -q
+
+    # Clone EI back-end and build war file
+    echo "Cloning Eiffel Intelligence back-end and building war file"
+    if [[ -d "eiffel-intelligence" ]]; then rm -Rf eiffel-intelligence; fi
+    git clone -q --depth=50 --branch=master https://github.com/eiffel-community/eiffel-intelligence.git
+    cd eiffel-intelligence
+    mvn package -DskipTests=true -q
+    cd ..
+}
+
+function do_start {
+    echo "Starting up Docker environment"
+
+    ## Set variables for Eiffel Intelligence war files.
+    export EI_FRONTEND_WAR_FILE=$(ls target/*.war)
+    export EI_BACKEND_WAR_FILE=$(cd eiffel-intelligence && ls target/*.war)
+
+    # Sets Docker image names and ports for containers.
+    # Also sets host variable
+    source src/main/docker/env.bash
+
+    # Set up docker containers and build the images of EI front-end, back-end and Jenkins
+    docker-compose -f src/main/docker/docker-compose.yml up -d --build
+
+    echo "Sleeping for 2 minutes, to let containers start up properly"
+    sleep 2m
+}
+
+function do_check {
+    echo "Checking if Jenkins container is up"
+
+    HTTP_CODE=503
+    until [[ $HTTP_CODE -eq "403" ]]; do
+        HTTP_CODE=$(curl -s -o /dev/null -w "%{http_code}\n" localhost:8082)
+        printf "."
+        sleep 5
+    done
+    printf " Jenkins container is up and running!\n"
+}
+
+function do_test {
+    echo "Starting system test"
+    # Set host variable
+    source src/main/docker/env.bash
+    mvn verify -P systemTest -Dei.frontend.url="http://${HOST}:8081" -Djenkins.external.url="http://${HOST}:8082" -B
+    STATUS=$?
+}
+
+function do_stop {
+    echo "Stopping Docker containers ..."
+    # Sourcing environment variables to avoid warnings
+    source src/main/docker/env.bash
+    docker-compose -f src/main/docker/docker-compose.yml down
+
+    # Cleaning up dangling docker images which were built using docker-compose command
+    docker rmi $(docker images -f "dangling=true" -q)
+}
+
+function usage {
+    cat <<EOF
+
+Usage: systemtest.sh COMMAND [OPTIONS]
+
+COMMANDS:
+    test :
+        First sets up Docker containers needed for system tests, then check Jenkins
+        container is healthy before starting the system tests defined for Eiffel
+        Intelligence, using maven verify command. After executing the tests, the
+        Docker containers are stopped and removed.
+    start :
```

Comment:
"start" command can be a bit confusing when script is named systemtest.
User might/commonly interpret "systemtest start" command as starting the tests, which is "test" command in this case.
Maybe rename "start" to e.g. "env-start" or something,
and rename "test" to "start", since script name is already named with "test" in the name.

Reply:
Hmm, I agree but maybe renaming the script file is the way to go then. Is "eiffel-intelligence.sh" too vague? The ommands start and test would have the correct meaning in that case. And both EI front-end and back-end with surrounding services are started so I'm reluctant to naming the script only "eiffel-intelligence-frontend"...

Reply:
toolbox.sh, entry.sh, entrypoint.sh, setup.sh, is some more suggestions.